src/TLuaInterpreter.cpp

```diff
@@ -3326,7 +3326,7 @@ int TLuaInterpreter::setTextFormat(lua_State* L)
         }
     }
 
-    if (windowName.isEmpty() || !windowName.compare(QStringLiteral("main"), Qt::CaseSensitive)) {
+    if (windowName.isEmpty() || windowName.compare(QStringLiteral("main"), Qt::CaseSensitive) == 0) {
```

Comment:
Are you saying that the condition here was wrong or just a little less clear that it could be?

In 'C' IIRC any numeric value that is NOT zero is a logical `true` so the original term, _including_ the negation was only true if the `windowName` is exactly `"main"` {note: at the time I put the `QString::compare(...)` in I expected the name of the "main" console to possibly be subjected to translation so the `QStringLiteral` was originally a `tr` method to accommodate translation.}

As far as I can tell this is not a functional change so I do not understand how this can be a bugfix? 😖

Reply:
If things are not working on anything except the "main" console window then the error is in the _other (else)_ branch of the `if(`...`) {`...`} else {`...`}`

Reply:
🤦‍♂️ Oh, ___now___ I get it - this one was okay but `setBold` `setItalics`, `setUnderline` and `setStrikeOut` were broken...